x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/action/TransportPutLifecycleAction.java

```diff
@@ -342,4 +344,18 @@ public Optional<String> reservedStateHandlerName() {
     public Set<String> modifiedKeys(PutLifecycleRequest request) {
         return Set.of(request.getPolicy().getName());
     }
+
+    private static class IlmLifecycleExecutor extends SimpleBatchedAckListenerTaskExecutor<UpdateLifecyclePolicyTask> {
+
+        @Override
+        public Tuple<ClusterState, ClusterStateAckListener> executeTask(UpdateLifecyclePolicyTask task, ClusterState clusterState) {
+            try {
+                return Tuple.tuple(task.execute(clusterState), task);
+            } catch (Exception e) {
+                throw new ElasticsearchException("failed to execute task", e);
+            }
```

Comment:
Is there a reason you added the try-catch block here? The `SimpleBatchedAckListenerTaskExecutor` already catches all exceptions and calls `onFailure` here:
https://github.com/elastic/elasticsearch/blob/a59c182f9f7e9d1bf3d6eecbc0e44f24ff91d053/server/src/main/java/org/elasticsearch/cluster/SimpleBatchedAckListenerTaskExecutor.java#L67-L69
It looks to me like that's what we want, but maybe I'm missing something.

Reply:
No just my IDE being fussy and not correctly adding the `throws` block to the method it generated for the interface -.- Have updated it now.